### PR TITLE
Handle possible errors during noreply commands

### PR DESCRIPTION
**Pull request by bisho**
## Motivation / Description
See: https://github.com/memcached/memcached/issues/868

TL;DR: Memcache noreply can return error responses on
write commands, defeating the purpose of the noreply
command can causing the response parsing on the socket
to become miss-aligned.

This adds no-op meta command to no-reply write requests.
We can return immediately without waiting
for the response (so there is no roundtrip penalty) and
keep a counter of the pending expected no-op responses.

The next non-noreply command will read and ignore
responses until the no-ops have been read, and proceed
to read the right response to the new command.

## Changes introduced
- Send no-op after noreply write commands
- Response reads will drop pending responses until the no-op
response when noops are expected in a connection.

### CODE REVIEW

File: src/meta_memcache/base/base_cache_pool.py [hunk 1]
```diff
@@ -182,13 +195,30 @@ def _conn_send_cmd(
...
         """
+        # No-reply flag behavior is hard to implement, so this
+        # is implemented client side, recording how many responses
+        # to ignore in client-side. See:
+        # https://github.com/memcached/memcached/issues/868
         if flags and Flag.NOREPLY in flags:
+            if command != MetaCommand.META_GET:
+                self._no_reply_cmds += 1
             return Success(flags=set([Flag.NOREPLY]))
+        while self._no_reply_cmds > 0:
```
Comment (ethervoid) [inline src/meta_memcache/base/base_cache_pool.py:212]: Who is possible to have `self._no_reply_cmds` > 0 if the part where is increased, line 210, finishes with a return statement?

Comment (bisho) [reply src/meta_memcache/base/base_cache_pool.py:212]: The first command that doesn't use `NOREPLY` will read the results for commands sent with NOREPLY and drop those, so it can get the reply it wants.

**Review by ethervoid — APPROVED** [review]
